images/base/Dockerfile

```diff
@@ -12,51 +12,53 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-# kind cluster base image
+# kind node base image
 #
 # For systemd + docker configuration used below, see the following references:
 # https://www.freedesktop.org/wiki/Software/systemd/ContainerInterface/
-# https://developers.redhat.com/blog/2014/05/05/running-systemd-within-docker-container/
-# https://developers.redhat.com/blog/2016/09/13/running-systemd-in-a-non-privileged-container/
 
-ARG BASE_IMAGE="ubuntu:19.10"
-FROM ${BASE_IMAGE}
+# start from ubuntu 19.10, this image is reasonably small as a starting point
+# for a kubernetes node image, it doesn't contain much we don't need
+FROM ubuntu:19.10
 
-# setting DEBIAN_FRONTEND=noninteractive stops some apt warnings, this is not 
-# a real argument, we're (ab)using ARG to get a temporary ENV again.
-ARG DEBIAN_FRONTEND=noninteractive
+# Configure containerd and runc binaries from kind-ci/containerd-nightlies repository
+# The repository contains latest stable releases and nightlies built for multiple architectures
+ARG CONTAINERD_VERSION="v1.3.0-7-g0b43a311"
+# Configure CNI binaries from upstream
+ARG CNI_VERSION="v0.8.2"
+# Configure crictl binary from upstream
+ARG CRICTL_VERSION="v1.16.1"
 
-COPY clean-install /usr/local/bin/clean-install
-RUN chmod +x /usr/local/bin/clean-install
+# copy in static files (configs, scripts)
+COPY files/ /
 
-# Get dependencies
-# The base image already has: ssh, apt, snapd
-# This is broken down into (each on a line):
+# Install dependencies, first from apt, then from release tarballs.
+# NOTE: we use one RUN to minimize layers.
+#
+# First we must ensure that our util scripts are executable.
+#
+# The base image already has: ssh, apt, snapd, but we need to install more packages.
```

Comment:
just curious now that you mention, should we add a TODO to try to reduce more the image size removing more packages?

Reply:
docker image layers don't work that way. a tombstone is added to the next layer if you delete something in a new layer.

Reply:
I was thinking that `--squash` was more mature

Reply:
squash is experimental in the daemon with no sign of that changing. experimental is problem because that means it is not on by default in the daemon, so we can't use it.